src/3rd_party_config.c

```diff
@@ -0,0 +1,223 @@
+/*
+ *   Software Updater - client side
+ *
+ *      Copyright © 2019 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "3rd_party_internal.h"
+#include "config.h"
+#include "lib/config_parser.h"
+#include "lib/list.h"
+#include "lib/sys.h"
+#include "swupd.h"
+#include <errno.h>
+#include <stdlib.h>
+
+#ifdef THIRDPARTY
+#define MAX_REPO_ALLOWED_FIELDS 2
+
+/** @brief Internal Memory representation of a repo.  */
+struct repo {
+	char *name;
+	char *url;
+};
+
+/** @brief List of repos. */
+struct list *repos_kv;
+
+/** @brief We populate this once every swupd init and use it many times. */
+static char *repo_config_file_path;
```

Comment:
So I did this way because we use the value several times during a single operation(third_party_config_create_file, write_repo_config etc). This means we will be running the below command several times to generate config_path. I felt it is efficient to run it once and store in a static global & use it several times.

```
char *repo_config_dir;
repo_config_dir = sys_path_join(globals.state_dir, globals.repo_config_dir);
repo_config_file_path = sys_path_join(repo_config_dir, globals.repo_config_file);
free_string(&repo_config_dir);
```

Reply:
Later I need to implement a sys_path_join that support adding multiple paths. For now you can use str_or_die() to create the path on one call.
The reason I don't like static variables is that it limits the usage of your code, makes it harder to read and harder to be reused.

Reply:
@otaviobp valid point. Will make the change.